### PR TITLE
Various Ansible optimizations

### PR DESCRIPTION
**Pull request by pat-s**
- Apply/check agent
- Ansible cfg optimizations
- use latest plugin version
- Move from `docker_compose` v1 to `docker_container` module since compose v1 is deprecated since mid 2023 and v2 does not support inline definitions
- Remove unused `docker_compose` role
- Move playbook to `playbooks/`

### CODE REVIEW

**Comment by pat-s** [comment]
ping @anbraten

File: .woodpecker/ansible.yaml [hunk 1]
```diff
@@ -6,39 +6,36 @@ when:
...
       galaxy: requirements.yaml
       inventory: inventory
       syntax_check: false
-      limit: ci_server
       become: true
       check: true
       user: root
       galaxy_force: false
-      vault_password:
+      vault-password:
```
Comment (anbraten) [inline .woodpecker/ansible.yaml:22]: Was this change intended?

File: playbooks/run.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        gather_subset:
+          - 'system'
+          - 'local'
+          - 'distribution'
+  roles:
+    - role: common
+    - role: docker
+    - role: caddy
+    - role: woodpecker_server
+    - role: woodpecker_agent
```
Comment (anbraten) [inline playbooks/run.yaml:15]:
```suggestion
    # - role: woodpecker_agent
```

We only use the autoscaler atm as the server is the smallest arm machine.

File: ansible.cfg [hunk 3]
```diff
@@ -1,17 +1,18 @@
...
-
 # Chick-Fil-A would like a word...
 nocows = True
-
 # Installs collections into [current dir]/ansible_collections/namespace/collection_name
 collections_path = ./
-
 roles_path = ./roles
-
-ask_vault_pass = False
```
Comment (anbraten) [inline ansible.cfg:11]: Was this change intended?

File: roles/weblate/tasks/main.yaml [hunk 4]
```diff
@@ -1,66 +1,73 @@
...
+      WEBLATE_EMAIL_HOST_PASSWORD: '{{ weblate_email_password | string }}'
+      WEBLATE_SERVER_EMAIL: '{{ weblate_email_username | string }}'
+      WEBLATE_DEFAULT_FROM_EMAIL: '{{ weblate_email_username | string }}'
+      WEBLATE_SITE_DOMAIN: '{{ weblate_domain | string }}'
+      WEBLATE_ADMIN_EMAIL: '{{ weblate_admin_email | string }}'
+      WEBLATE_SITE_TITLE: '{{ weblate_site_title | string }}'
+      POSTGRES_USER: 'weblate'
+      POSTGRES_PASSWORD: '{{ weblate_postgres_password | string }}'
+      POSTGRES_DATABASE: 'weblate'
+      POSTGRES_HOST: 'database'
```
Comment (anbraten) [inline roles/weblate/tasks/main.yaml:63]:
```suggestion
      POSTGRES_HOST: 'weblate_database'
```

File: roles/weblate/tasks/main.yaml [hunk 5]
```diff
@@ -1,66 +1,73 @@
...
+      POSTGRES_PORT: '5432'
+      REDIS_HOST: 'cache'
```
Comment (anbraten) [inline roles/weblate/tasks/main.yaml:65]:
```suggestion
      REDIS_HOST: 'weblate_cache'
```

File: roles/woodpecker_server/tasks/main.yaml [hunk 6]
```diff
@@ -6,35 +6,29 @@
...
+    restart: true
+    restart_policy: 'always'
+    volumes:
+      - /opt/woodpecker:/var/lib/woodpecker/
+    networks:
+      - name: web
+    env:
+      WOODPECKER_OPEN: 'true'
+      WOODPECKER_HOST: '{{ woodpecker_server_host | string }}'
+      WOODPECKER_ORGS: '{{ woodpecker_server_orgs| string }}'
```
Comment (anbraten) [inline roles/woodpecker_server/tasks/main.yaml:24]:
```suggestion
      WOODPECKER_ORGS: '{{ woodpecker_server_orgs | string }}'
```

File: roles/woodpecker_server/tasks/main.yaml [hunk 7]
```diff
@@ -6,35 +6,29 @@
...
+      WOODPECKER_ADMIN: '{{ woodpecker_server_admins| string }}'
```
Comment (anbraten) [inline roles/woodpecker_server/tasks/main.yaml:25]:
```suggestion
      WOODPECKER_ADMIN: '{{ woodpecker_server_admins | string }}'
```

File: roles/woodpecker_server/tasks/main.yaml [hunk 8]
```diff
@@ -6,35 +6,29 @@
...
+      WOODPECKER_GITHUB: 'true'
+      WOODPECKER_GITHUB_CLIENT: '{{ woodpecker_server_github_client| string }}'
+      WOODPECKER_GITHUB_SECRET: '{{ woodpecker_server_github_secret| string }}'
+      WOODPECKER_AGENT_SECRET: '{{ woodpecker_server_agent_secret| string }}'
```
Comment (anbraten) [inline roles/woodpecker_server/tasks/main.yaml:29]:
```suggestion
      WOODPECKER_GITHUB_CLIENT: '{{ woodpecker_server_github_client | string }}'
      WOODPECKER_GITHUB_SECRET: '{{ woodpecker_server_github_secret | string }}'
      WOODPECKER_AGENT_SECRET: '{{ woodpecker_server_agent_secret | string }}'
```

**Review by anbraten — COMMENTED** [review]
Before applying we definitely should do a snapshot 😉

File: roles/woodpecker_server/tasks/main.yaml [hunk 9]
```diff
@@ -6,35 +6,29 @@
...
-            - WOODPECKER_AGENT_SECRET={{ woodpecker_server_agent_secret }}
-            - WOODPECKER_GITHUB_MERGE_REF=false
-      networks:
-        web:
-          external:
-            name: web
+- name: Woodpecker server container
+  community.docker.docker_container:
+    name: woodpecker_server
+    hostname: woodpecker_server
```
Comment (anbraten) [inline roles/woodpecker_server/tasks/main.yaml:12]:
```suggestion
    hostname: woodpecker-server
```

Has to be written like this for caddy

File: site.yaml [hunk 10]
```diff
@@ -1,31 +0,0 @@
-# - name: Prepare Hetzner Cloud
```
Comment (anbraten) [inline site.yaml:1]: Readme needs to be updated

File: .woodpecker/ansible.yaml (hunk 1 above)
Comment (pat-s) [reply .woodpecker/ansible.yaml:27]: Was about to harmonize everything to `_`. Sanitation makes both work the same way.

File: ansible.cfg (hunk 3 above)
Comment (pat-s) [reply ansible.cfg:11]: Can't remember fully. If it is needed, keep it. Is it needed actually?

Comment (pat-s) [reply ansible.cfg:11]: The default is false, so we can also remove it: https://docs.ansible.com/ansible/latest/reference_appendices/config.html#default-ask-vault-pass

File: site.yaml (hunk 10 above)
Comment (pat-s) [reply site.yaml:1]: done

File: .woodpecker/ansible.yaml (hunk 1 above)
Comment (anbraten) [reply .woodpecker/ansible.yaml:27]: In that case we should stick with `_` I guess?

```suggestion
      vault_password:
```
